src/App.js

```diff
@@ -3,21 +3,31 @@ import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 import { useState, useEffect } from "react";
 
-// /*This is the app that will display the arrays of Object and the form with button*
 
-const useSemiPersistentState = () => {
+
+function App() {
   const [todoList, setTodoList] = useState(
     JSON.parse(localStorage.getItem("savedTodoList")) || []
   );
 
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]); 
-    return [todoList, setTodoList];
-};
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList: todoList } });
```

Comment:
And then here is where you'll do the actual data fetching. This technically works the same way as how you currently have it, but the spirit of the exercise to mimic how you would set up an async function to fetch data from an external source.
```suggestion
        resolve({ data: { todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [], } });
```